src/content/migrate-from-hardhat2/index.md

```diff
@@ -7,4 +7,314 @@ description: How to migrate from Hardhat 2 to Hardhat 3
 
 ## Overview
 
-Hardhat 3 is a complete rewrite of Hardhat 2. Here's how to migrate.
+Hardhat 3 is a complete rewrite of Hardhat 2. While many features are familiar, several fundamental changes mean the new version is not compatible with Hardhat 2 projects out of the box:
+
+- **ESM-first**: Your Hardhat config must be an ES module. Scripts and JavaScript/TypeScript tests can still be CommonJS, but ESM is the default.
+- **Declarative config**: Plugins, tasks, and other extensions are configured explicitly in your config file instead of being registered by side effects.
```

Comment:
```suggestion
- **Declarative config**: Plugins, tasks, and other extensions are configured explicitly in your config instead of being registered by side effects.
```